README.md

```diff
@@ -44,32 +44,84 @@ import smartapp from '@smartthings/smartapp'
+    .page('mainPage', (context, page, configData) => {
+        page.section('sensors', section => {
+           section.deviceSetting('contactSensor').capabilities(['contactSensor']).required(false);
         });
-        page.section('lights', (section) => {
+        page.section('lights', section => {
             section.deviceSetting('lights').capabilities(['switch']).multiple(true).permissions('rx');
         });
     })
-    .updated(() => {
-        smartapp.api.devices.unsubscribeAll().then(() => {
-            smartapp.api.devices.subscribe(smartapp.config.contactSensor, 'contactSensor', 'contact', 'openCloseHandler');
+    .installed((context, installData) => {
+        console.log('installed', JSON.stringify(installData));
+    })
+    .uninstalled((context, uninstallData) => {
+        console.log('uninstalled', JSON.stringify(uninstallData));
+    })
+    .updated((context, updateData) => {
+        console.log('updated', JSON.stringify(updateData));
+        context.api.subscriptions.unsubscribeAll().then(() => {
+              console.log('unsubscribeAll() executed');
+              context.api.subscriptions.subscribeToDevices(context.config.contactSensor, 'contactSensor', 'contact', 'myDeviceEventHandler');
         });
     })
-    .subscribedEventHandler('openCloseHandler', (event) => {
-        const value = event.value === 'open' ? 'on' : 'off';
-        smartapp.api.devices.sendCommands(smartapp.config.lights, 'switch', value);
+    .subscribedEventHandler('myDeviceEventHandler', (context, deviceEvent) => {
+        const value = deviceEvent.value === 'open' ? 'on' : 'off';
+        context.api.devices.sendCommands(context.config.lights, 'switch', value);
+        console.log(`sendCommands(${JSON.stringify(context.config.lights)}, 'switch', '${value}')`);
+
+        /* All subscription event handler types:
+         *   - DEVICE_EVENT (context, deviceEvent)
+         *   - TIMER_EVENT (context, timerEvent)
+         *   - DEVICE_COMMANDS_EVENT (context, deviceId, command, deviceCommandsEvent)
+         *   - MODE_EVENT (context, modeEvent)
+         *   - SECURITY_ARM_STATE_EVENT (context, securityArmStateEvent)
+        */
     });
-exports.handle = (evt, context, callback) => {
-    smartapp.handleLambdaCallback(evt, context, callback);
+
+/* GET home page. */
+server.post('/', function(req, res, next) {
+  console.log(`${new Date().toISOString()} ${req.method} ${req.path} ${req.body && req.body.lifecycle}`);
+  smartapp.handleHttpCallback(req, res);
+});
+
+/* Start listening at your defined PORT */
+server.listen(PORT, () => console.log(`Server is up and running on port ${PORT}`));
+```
+
+### Running as an AWS Lambda function
+
+To run as a Lambda function instead of an HTTP server, ensure that your main entry file exports `smartapp.handleLambdaCallback(...)`.
+
+> **Note:** This snippet is heavily truncated for brevity – see the web service example above a more detailed example of how to define a `smartapp`.
```

Comment:
Why not have fully functional copy-pasteable for both lambda and web server?

Reply:
It would be harder to see the functional differences if the Lambda example was code-complete.

Reply:
I suppose. In my mind I'd be wanting to quickly get something running, so would want copy/paste code (like I did) to get something running in Lambda. The functional differences could be called out in text before the code block. That said there's enough good here to deal with that later, so I won't hold the PR for it.

Reply:
I think it's Ok to inline but needs more code comments that give clarity for the examples.

Reply:
I agree that, generally speaking, there is still room for improvement in this readme. My main goal is just to create copy-pastable code that 100% will work for users, and the web service example does function as-is. The only caveat is that they need to pay attention to the public key assignment, which needs to exist, but I commented it.

Ty for reviews on short notice